django/db/models/fields/related.py

```diff
@@ -1031,6 +1031,10 @@ def save_form_data(self, instance, data):
             setattr(instance, self.name, data)
         else:
             setattr(instance, self.attname, data)
+            # remote field object must be cleared otherwise Model.save()
+            # will reassign attname using the related object pk
+            if data is None:
+                setattr(instance, self.name, data)
```

Comment:
All tests are passing without this change. Can we add test to cover that?

Reply:
I removed this block of code and ran all tests. The following case is failing:
```
FAIL: test_assignment_of_none (model_forms.tests.ModelOneToOneFieldTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.7/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.7/unittest/case.py", line 615, in run
    testMethod()
  File "/home/user/code/django-project/django/tests/model_forms/tests.py", line 1887, in test_assignment_of_none
    self.assertIsNone(new_author.publication)
  File "/usr/lib/python3.7/unittest/case.py", line 1250, in assertIsNone
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python3.7/unittest/case.py", line 680, in fail
    raise self.failureException(msg)
AssertionError: <Publication: Pravda> is not None
```
Sorry if I'm misunderstanding something. Would the failed testcase above be suffice in covering this logic?

Reply:
Yes, sounds good, thanks :+1:  I checked only new tests.